pkg/networkserver/downlink.go

```diff
@@ -135,7 +135,15 @@ func (ns *NetworkServer) nextDataDownlinkTaskAt(ctx context.Context, dev *ttnpb.
 		log.FromContext(ctx).WithError(err).Warn("Failed to determine device band")
 		return time.Time{}, nil
 	}
-	slot, ok := nextDataDownlinkSlot(ctx, dev, phy, ns.defaultMACSettings, earliestAt)
+	profile := &ttnpb.MACSettingsProfile{}
+	if dev.MacSettingsProfileIds != nil {
+		profile, err = ns.macSettingsProfiles.Get(ctx, dev.GetMacSettingsProfileIds(), []string{"mac_settings"})
```

Comment:
Is there a reason to pass a non-nil empty `*ttnpb.MACSettingsProfile`?

It looks like you're consistently using getters so it is safe to pass `nil`.

The reason I'm asking is that the empty `*ttnpb.MACSettingsProfile` is not valid, as `mac_settings` is a required field but it is `nil`. Our future selves may assume that it is safe to deference `mac_settings`, while it is not.

Consider:

```suggestion
	var profile *ttnpb.MACSettingsProfile
	if dev.MacSettingsProfileIds != nil {
		profile, err = ns.macSettingsProfiles.Get(ctx, dev.GetMacSettingsProfileIds(), []string{"mac_settings"})
```

Reply:
Also is there a reason to pass `*ttnpb.MACSettingsProfile` instead of just `*ttnpb.MACSettings`?

Reply:
> Is there a reason to pass a non-nil empty `*ttnpb.MACSettingsProfile`?
> 

I was not sure to safe to pass `nil` but as you said getters are used everywhere so it should be fine. I've changed it.

> The reason I'm asking is that the empty `*ttnpb.MACSettingsProfile` is not valid, as `mac_settings` is a required field but it is `nil`. Our future selves may assume that it is safe to deference `mac_settings`, while it is not.
> 

Good point.

Reply:
> Also is there a reason to pass `*ttnpb.MACSettingsProfile` instead of just `*ttnpb.MACSettings`?

Not really, only the MAC settings are used from the profile, so it should be enough to pass only that. I've changed it.